gen/emit/EmitJulia.cpp

```diff
@@ -70,47 +70,91 @@ struct EmitJulia : public Emitter
         }
     }
 
+    string topLevelPackage(const string& package) {
+        auto packageParts = StringUtil::split(package, '.');
+        if (packageParts.size() > 0) {
+            return packageParts.at(0);
+        } else {
+            return "";
+        }
+    }
+
     void emitDependencies()
     {
-        unordered_set<string> dependencies;
+        unordered_set<string> imports;
+        string lsPackage = topLevelPackage(ls.structname.package);
+        if (lsPackage != "") {
+            imports.insert("import " + lsPackage);
+        }
+
         for (auto& lm : ls.members) {
             auto& tn = lm.type.fullname;
             if (!ZCMGen::isPrimitiveType(tn) &&
-                dependencies.find(tn) == dependencies.end() &&
+                imports.find(tn) == imports.end() &&
                 tn != ls.structname.fullname) {
```

Comment:
I don't believe zcmtypes are allowed to contain themselves, and if that is actually allowed, it shouldn't be. We should prevent this sort of error at a higher level.

Reply:
You actually can have a type that contains itself. I believe the type just has to be an array. That way you can break the infinite recursion by having a base case where the array is size 0

Reply:
Unfortunately, you can't have a Julia type that contains a field with its own type (though you can have an un-typed field, at the cost of some performance)

Reply:
Err, never mind, I was incorrect. You *can* have a Julia type that contains itself.